### PR TITLE
add support for lines

### PR DESCRIPTION
**Pull request by rokit**
This adds support for lines, which partially addresses #34. An optional texture index is supported based on [this doc](https://paulbourke.net/dataformats/obj/), since apparently some applications can make use of that.

### CODE REVIEW

File: src/obj.rs
Comment (mororo18) [inline src/obj.rs:1]: [src/obj.rs:841-843] 
```rust 
// we found a new material that was applied to an existing
// non-empty object. It is treated as a new group.
if g.material.is_some() && !g.polys.is_empty() {
```

i think you need to check here if  `!g.polys.is_empty() || !g.lines.is_empty()`

Comment (rokit) [reply src/obj.rs:1]: Good catch. Updated.

**Review by kvark — APPROVED** [review]

**Comment by kvark** [comment]
I'm not really reviewing this in good detail. If anybody is interested in reviewing incoming PRs - I'm happy to grant them powers.

**Comment by rokit** [comment]
I rebased and fixed the conflict.

**Pull request merged**